samples/ab_testing/plugin.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_ab_testing]
+#include <ctime>
+#include <string>
+
+#include "proxy_wasm_intrinsics.h"
+
+class MyHttpContext : public Context {
+ public:
+  explicit MyHttpContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t headers,
+                                       bool end_of_stream) override {
+    std::string path = getRequestHeader(":path")->toString();
+
+    const std::string default_file = "/file1.blah";
+    const std::string experiment_file = "/file2.blah";
+    if (path.length() >= default_file.length() &&
+        std::equal(default_file.rbegin(), default_file.rend(), path.rbegin())) {
```

Comment:
I got a little lost in this logic -- is it a substring check on the full path? Worth a comment.,

You might also consider using URL parsing (and manipulation below) from boost, which I am adding in PR #17.

Reply:
I have mode some refactoring here by moving this if condition to a separate function and adding around some comments. Please, take a look if it's more readable or whether you would like me to improve it further.

As for using the Boost library: I have a mild preference to have as few dependecies as we can get. I think it's more beneficial to focus on a specific thing that we are trying to showcase with this sample.